```
error[E0432]: unresolved import `alloc::sync`
  --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/scale-info-2.7.0/src/prelude.rs:50:13
   |
50 |             sync,
   |             ^^^^ no `sync` in the root
   |
   = help: consider importing this module instead:
           core::sync
```

Interesting; I wouldn't expect `scale-info` to have a no-std related issue.

For alloc::sync the docs say:

```
Note: This module is only available on platforms that support atomic loads and stores of pointers. This may be detected at compile time using #[cfg(target_has_atomic = "ptr")].
```

So maybe the no_std target I suggested has this limitation too. Maybe there's a better target choice?

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -95,7 +95,7 @@ jobs:
         with:
           profile: minimal
           toolchain: stable
-          target: thumbv6m-none-eabi
+          target: aarch64-unknown-none
           override: true
 
       - name: Rust Cache

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -105,9 +105,9 @@ jobs:
         uses: actions-rs/cargo@v1.0.3
         with:
           command: check
-          # The thumbv6m-none-eabi doesn't support `std`, so this
+          # The aarch64-unknown-none doesn't support `std`, so this
           # will fail if the crate is not no_std compatible.
-          args: --no-default-features --target thumbv6m-none-eabi --features scale-info,serde
+          args: --no-default-features --target aarch64-unknown-none --features scale-info,serde
 
 
   fmt:

